odata_client/src/main/java/com/sdl/odata/client/AbstractODataClientComponentsProvider.java

```diff
@@ -47,11 +48,34 @@
 
     public AbstractODataClientComponentsProvider(Iterable<String> edmEntityClasses, Properties properties) {
         webServiceUri = getServiceUri(properties);
-        endpointCaller = new BasicEndpointCaller(properties);
+        endpointCaller = initializeEndpointCaller(properties);
 
         initComponentsProvider(edmEntityClasses);
     }
 
+    /**
+     * If the TracingEndpointCaller class is in a classpath - use it, otherwise use BasicEndpointCaller.
+     *
+     * @param properties properties argument for endpoint caller
+     * @return endpoint caller instance
+     */
+    public static EndpointCaller initializeEndpointCaller(Properties properties) {
+        EndpointCaller ec;
+        try {
+            LOG.debug("Initializing endpoint caller. " +
+                    "Checking whether 'com.sdl.odata.client.caller.TracingEndpointCaller' is in classpath.");
+            Class<?> tracingEndpointCallerClass = Class.forName("com.sdl.odata.client.caller.TracingEndpointCaller");
```

Comment:
can we have this - 'com.sdl.odata.client.caller.TracingEndpointCaller' as static constant ?